src/posix/ot_instance.c

```diff
@@ -0,0 +1,323 @@
+
+#include <common/code_utils.hpp>
+#include <lib/platform/exit_code.h>
+#include <openthread/openthread-system.h>
+#include <openthread/platform/misc.h>
+
+#define MULTIPLE_INSTANCE_MAX 10
+#define MAX_TTY_LEN 100
+#define MAX_RADIO_URL_LEN 200
+#define MAX_INTERFACE_LEN 100
+
+static otInstance *gInstance = NULL;
+static pthread_t   gThreadId;
+pthread_mutex_t    gLock;
+bool               gTerminate = 0;
+
+typedef struct Param
+{
+    char     comPort[MAX_TTY_LEN];
+    uint16_t debugLevel;
+} Param;
+
+typedef struct PosixConfig
+{
+    otPlatformConfig mPlatformConfig;    ///< Platform configuration.
+    otLogLevel       mLogLevel;          ///< Debug level of logging.
+    bool             mPrintRadioVersion; ///< Whether to print radio firmware version.
+    bool             mIsVerbose;         ///< Whether to print log to stderr.
+} PosixConfig;
+
+static otInstance *InitInstance(PosixConfig *aConfig)
+{
+    otInstance *instance = NULL;
+
+    syslog(LOG_INFO, "Running %s", otGetVersionString());
+    syslog(LOG_INFO, "Thread version: %hu", otThreadGetVersion());
+    IgnoreError(otLoggingSetLevel(aConfig->mLogLevel));
+
+    instance = otSysInit(&aConfig->mPlatformConfig);
+    VerifyOrDie(instance != NULL, OT_EXIT_FAILURE);
+    syslog(LOG_INFO, "Thread interface: %s", otSysGetThreadNetifName());
+
+    if (aConfig->mPrintRadioVersion)
+    {
+        printf("%s\n", otPlatRadioGetVersionString(instance));
+    }
+    else
+    {
+        syslog(LOG_INFO, "RCP version: %s", otPlatRadioGetVersionString(instance));
+    }
+
+    if (aConfig->mPlatformConfig.mDryRun)
+    {
+        exit(OT_EXIT_SUCCESS);
+    }
+
+    return instance;
+}
+
+void otPlatReset(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+
+    otSysDeinit();
+
+    assert(false);
+}
+
+static bool getRadioURL(char *comPort)
+{
+    struct stat st;
+    char        radioDevice[MAX_RADIO_URL_LEN] = {0};
+
+    /* For multiple devices and intefaces,
+     * we will align the index of them.
+     */
+    sprintf(radioDevice, "/dev/%s", comPort);
+    if (stat(radioDevice, &st) == 0)
+    {
+        syslog(LOG_INFO, "radio device found [%s]", radioDevice);
+        return true;
+    }
+    return false;
+}
+
+static int getInterface(void)
+{
+    struct stat st;
+    int         i                        = 0;
+    char        iface[MAX_INTERFACE_LEN] = {0};
+
+    for (i = 0; i < MULTIPLE_INSTANCE_MAX; i++)
+    {
+        sprintf(iface, "/sys/class/net/wpan%d", i);
+        if (stat(iface, &st) == 0)
+        {
+            syslog(LOG_INFO, "Interface is already used [%s]", iface);
+            continue;
+        }
+        else
+        {
+            syslog(LOG_INFO, "find empty interface [%s]", iface);
+            break;
+        }
+    }
+    if (i == MULTIPLE_INSTANCE_MAX)
+    {
+        syslog(LOG_CRIT, "Interface reached max count...Not able to create new inteface");
+        return -1;
+    }
+
+    return i;
+}
+
+void otCreateInstance(void *arg)
```

Comment:
We generally use `ot` prefix for OT public or platform APIs. 

Are these intended as new (posix-specific) platform APIs?
- If so, suggest having them declared and documented in a posix-specific platform header (similar to  `openthread-system.h`).
- If not, then suggest renaming and not using `ot` prefix?

Reply:
Updated the APIs as per your comments